File: Arrays/P008.py
def brute(arr):
    n=len(arr)
    for i in range(n+1):
        c=0
        for j in range(n):
            if arr[j]==i:
                c=1
        if c==0:
            return i
    return -1

def optimal(arr):
    n=len(arr)
    tot=n*(n+1)//2
    suma=0
    for i in range(n):
        suma+=arr[i]
    return tot-suma

if __name__=='__main__':
    arr=input("Enter array").split()
    nums=list(map(int,arr))
    print("Brute: ",brute(nums))
    print('Optimal: ',optimal(nums))

File: Striver_C++_Problems/DP/P008.cpp
// P008.cpp - Find missing number in range [0..n]

#include <bits/stdc++.h>
using namespace std;

int brute(const vector<int>& arr) {
    int n = arr.size();
    for (int v = 0; v <= n; v++) {
        bool found = false;
        for (int x : arr) {
            if (x == v) {
                found = true;
                break;
            }
        }
        if (!found) return v;
    }
    return -1;
}

int optimal(const vector<int>& arr) {
    int n = arr.size();
    long long tot = 1LL * n * (n + 1) / 2;
    long long suma = 0;
    for (int x : arr) suma += x;
    return (int)(tot - suma);
}

int main() {
    cout << "Enter array: ";
    string line;
    getline(cin, line);
    stringstream ss(line);
    vector<int> nums;
    int x;
    while (ss >> x) nums.push_back(x);

    cout << "Brute: " << brute(nums) << "\n";
    cout << "Optimal: " << optimal(nums) << "\n";
    return 0;
}
